src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -227,125 +226,120 @@ library BeaconChainProofs {
      * @notice This function verifies the latestBlockHeader against the state root. the latestBlockHeader is 
      * a tracked in the beacon state.
      * @param beaconStateRoot is the beacon chain state root to be proven against.
-     * @param latestBlockHeaderProof is the provided merkle proof
-     * @param latestBlockHeaderRoot is hashtree root of the latest block header in the beacon state
+     * @param stateRootProof is the provided merkle proof
+     * @param latestBlockRoot is hashtree root of the latest block header in the beacon state
      */
-    function verifyStateRootAgainstLatestBlockHeaderRoot(
+    function verifyStateRootAgainstLatestBlockRoot(
         bytes32 beaconStateRoot,
-        bytes32 latestBlockHeaderRoot,
-        bytes calldata latestBlockHeaderProof
+        bytes32 latestBlockRoot,
+        bytes calldata stateRootProof
     ) internal view {
-        require(latestBlockHeaderProof.length == 32 * (BEACON_BLOCK_HEADER_FIELD_TREE_HEIGHT),
-            "BeaconChainProofs.verifyStateRootAgainstLatestBlockHeaderRoot: Proof has incorrect length");
-        //Next we verify the slot against the blockHeaderRoot
-        require(Merkle.verifyInclusionSha256({proof: latestBlockHeaderProof, root: latestBlockHeaderRoot, leaf: beaconStateRoot, index: STATE_ROOT_INDEX}),
-            "BeaconChainProofs.verifyStateRootAgainstLatestBlockHeaderRoot: Invalid latest block header root merkle proof");
+        require(stateRootProof.length == 32 * (BEACON_BLOCK_HEADER_FIELD_TREE_HEIGHT),
+            "BeaconChainProofs.verifyStateRootAgainstLatestBlockRoot: Proof has incorrect length");
+        //Next we verify the slot against the blockRoot
+        require(Merkle.verifyInclusionSha256({proof: stateRootProof, root: latestBlockRoot, leaf: beaconStateRoot, index: STATE_ROOT_INDEX}),
+            "BeaconChainProofs.verifyStateRootAgainstLatestBlockRoot: Invalid latest block header root merkle proof");
     }
 
     /**
      * @notice This function verifies the slot and the withdrawal fields for a given withdrawal
-     * @param beaconStateRoot is the beacon chain state root to be proven against.
-     * @param withdrawalProofs is the provided set of merkle proofs
+     * @param withdrawalProof is the provided set of merkle proofs
      * @param withdrawalFields is the serialized withdrawal container to be proven
      */
-    function verifyWithdrawalProofs(
-        bytes32 beaconStateRoot,
```

Comment:
I always like passing the root separately because it feels like its "separate" from the leaf/proofs

Reply:
gas doesn't care about your feelings

Reply:
so the optimization here is adding it to the struct?  Why does that save gas?